### ISSUE TITLE
loss的值在训练一段时间后不下降了，是那些因素引起的啊？

### ISSUE THREAD

**Issue by lazywl** [MoyanZitto/keras-cn#13]
Epoch 9/12
  66/3300 [..............................] - ETA: 187s - loss: 1.3111 - acc: 0.6
 132/3300 [>.............................] - ETA: 183s - loss: 1.3862 - acc: 0.6
 198/3300 [>.............................] - ETA: 180s - loss: 1.3752 - acc: 0.6
 264/3300 [=>............................] - ETA: 176s - loss: 1.3745 - acc: 0.6
 330/3300 [==>...........................] - ETA: 172s - loss: 1.3791 - acc: 0.6
 396/3300 [==>...........................] - ETA: 168s - loss: 1.3531 - acc: 0.6
 462/3300 [===>..........................] - ETA: 164s - loss: 1.4262 - acc: 0.5
 528/3300 [===>..........................] - ETA: 161s - loss: 1.4514 - acc: 0.5
 594/3300 [====>.........................] - ETA: 157s - loss: 1.4463 - acc: 0.5
 660/3300 [=====>........................] - ETA: 153s - loss: 1.4639 - acc: 0.5
 726/3300 [=====>........................] - ETA: 149s - loss: 1.4817 - acc: 0.5
 792/3300 [======>.......................] - ETA: 145s - loss: 1.5080 - acc: 0.5
 858/3300 [======>.......................] - ETA: 141s - loss: 1.5133 - acc: 0.5
 924/3300 [=======>......................] - ETA: 138s - loss: 1.4968 - acc: 0.5
 990/3300 [========>.....................] - ETA: 134s - loss: 1.5090 - acc: 0.5
1056/3300 [========>.....................] - ETA: 130s - loss: 1.5078 - acc: 0.5
1122/3300 [=========>....................] - ETA: 126s - loss: 1.4920 - acc: 0.5
1188/3300 [=========>....................] - ETA: 122s - loss: 1.4766 - acc: 0.5
1254/3300 [==========>...................] - ETA: 118s - loss: 1.4613 - acc: 0.5
1320/3300 [===========>..................] - ETA: 115s - loss: 1.4620 - acc: 0.5
1386/3300 [===========>..................] - ETA: 111s - loss: 1.4687 - acc: 0.5
1452/3300 [============>.................] - ETA: 107s - loss: 1.4600 - acc: 0.5
1518/3300 [============>.................] - ETA: 103s - loss: 1.4676 - acc: 0.5
1584/3300 [=============>................] - ETA: 99s - loss: 1.4661 - acc: 0.57
1650/3300 [==============>...............] - ETA: 96s - loss: 1.4704 - acc: 0.57
1716/3300 [==============>...............] - ETA: 92s - loss: 1.4654 - acc: 0.57
1782/3300 [===============>..............] - ETA: 88s - loss: 1.4604 - acc: 0.57
1848/3300 [===============>..............] - ETA: 84s - loss: 1.4622 - acc: 0.56
1914/3300 [================>.............] - ETA: 80s - loss: 1.4664 - acc: 0.56
1980/3300 [=================>............] - ETA: 76s - loss: 1.4751 - acc: 0.56
2046/3300 [=================>............] - ETA: 72s - loss: 1.4672 - acc: 0.56
2112/3300 [==================>...........] - ETA: 69s - loss: 1.4711 - acc: 0.56
2178/3300 [==================>...........] - ETA: 65s - loss: 1.4716 - acc: 0.56
2244/3300 [===================>..........] - ETA: 61s - loss: 1.4764 - acc: 0.55
2310/3300 [====================>.........] - ETA: 57s - loss: 1.4698 - acc: 0.55
2376/3300 [====================>.........] - ETA: 53s - loss: 1.4797 - acc: 0.55
2442/3300 [=====================>........] - ETA: 49s - loss: 1.4791 - acc: 0.56
2508/3300 [=====================>........] - ETA: 46s - loss: 1.4839 - acc: 0.55
2574/3300 [======================>.......] - ETA: 42s - loss: 1.4884 - acc: 0.55
2640/3300 [=======================>......] - ETA: 38s - loss: 1.4848 - acc: 0.55
2706/3300 [=======================>......] - ETA: 34s - loss: 1.4841 - acc: 0.55
2772/3300 [========================>.....] - ETA: 30s - loss: 1.4817 - acc: 0.55
2838/3300 [========================>.....] - ETA: 26s - loss: 1.4770 - acc: 0.56
2904/3300 [=========================>....] - ETA: 23s - loss: 1.4796 - acc: 0.55
2970/3300 [==========================>...] - ETA: 19s - loss: 1.4786 - acc: 0.56
3036/3300 [==========================>...] - ETA: 15s - loss: 1.4703 - acc: 0.56
3102/3300 [===========================>..] - ETA: 11s - loss: 1.4700 - acc: 0.56

**Comment by MoyanZitto** [comment]
同学，你这不是loss不下降，而是loss在上升acc在下降啊= =
把具体的问题和网络详情拿出来看一下

**Comment by lazywl** [comment]
# -_\- coding: utf-8 -_-

"""
Created on Thu Oct 13 12:56:24 2016

@author:
做手写字母的识别
"""
import numpy as np
from LoadMyData import LoadData
np.random.seed(1337) # for reproducibililty

from keras.models import Sequential
from keras.layers.core import Dense, Dropout, Activation, Flatten
from keras.layers.convolutional import Convolution2D, MaxPooling2D
from keras.utils import np_utils
from keras.optimizers import adadelta

batch_size = 66
nb_classes = 33
nb_epoch = 12

# input image dimensions

img_rows, img_cols = 28 , 28

# number of convolutional filters to use

nb_filters = 32

# size of pooling area for max pooling

nb_pool = 2

# convolution kernel size

nb_conv = 3

def create_model():
    model = Sequential()
    # 32 convolution filters , the size of convolution kernel is 3 \* 3
    # border_mode can be 'valid' or 'full'

```
# 'full'  zero-pads image to multiple of filter shape to generate output of shape: image_shape + filter_shape - 1
# when used as the first layer, you should specify the shape of inputs 
# the first number means the channel of an input image, 1 stands for grayscale imgs, 3 for RGB imgs
model.add(Convolution2D(nb_filters, nb_conv, nb_conv,
                        border_mode='valid',
                        input_shape=(1, img_rows, img_cols)))
# use rectifier linear units : max(0.0, x)
model.add(Activation('relu'))


# second convolution layer with 32 filters of size 3*3
model.add(Convolution2D(nb_filters, nb_conv, nb_conv))
model.add(Activation('relu'))
# max pooling layer, pool size is 2 * 2
model.add(MaxPooling2D(pool_size=(nb_pool, nb_pool)))
# drop out of max-pooling layer , drop out rate is 0.25 
model.add(Dropout(0.25))

model.add(Flatten())
# add fully connected layer with 128 hidden units
model.add(Dense(128))
model.add(Activation('relu'))
model.add(Dropout(0.5))
# output layer with softmax 
model.add(Dense(nb_classes))
model.add(Activation('softmax'))
# use cross-entropy cost and adadelta to optimize params
#Adadelta = adadelta(lr=0.1, rho=0.95, epsilon=1e-06)
model.compile(loss='categorical_crossentropy', optimizer='adadelta',metrics=['accuracy'])
# train model with bath_size =10, epoch=12
# set verbose=1 to show train info
return model
```

X_train , Y_train = LoadData()
print X_train.shape[0], 'train samples'
Y_train = np_utils.to_categorical(Y_train, nb_classes)
model = create_model()
model.fit(X_train, Y_train, batch_size=batch_size, nb_epoch=nb_epoch,
      verbose=1,shuffle=True)
model.save("MyCNN_model3.h5")
#model.fit(X_train,Y_train,batch_size=batch_size, nb_epoch=nb_epoch)

# evaluate on test set

#score = model.evaluate(X_test, Y_test)
#print('Test accuracy:', score[1])

**Comment by MoyanZitto** [comment]
@lazywl 
程序没啥问题，注释太乱，一大堆没有必要的注释。另外github上有粘贴代码的方式，最好不要直接把你的plain代码贴过来，如这样：

``` python
X_train , Y_train = LoadData()
print X_train.shape[0], 'train samples'
Y_train = np_utils.to_categorical(Y_train, nb_classes)
model = create_model()
model.fit(X_train, Y_train, batch_size=batch_size, nb_epoch=nb_epoch,
verbose=1,shuffle=True)
model.save("MyCNN_model3.h5")
#model.fit(X_train,Y_train,batch_size=batch_size, nb_epoch=nb_epoch)
```

可以单机编辑栏上面的“Preview”来预览，单击下面的“Styling with markdown is supported”或者搜索markdown教程获取基本的帮助。

网络主要问题可能是太浅，学习能力不够，但代码上问题不大，仔细调参吧

**Comment by lazywl** [comment]
@MoyanZitto  谢谢了，我再调调。
